tools/c7n_azure/c7n_azure/lookup.py

```diff
@@ -20,19 +20,18 @@ class Lookup(object):
 
     schema = {
         'type': 'object',
-        'properties': {
-            'oneOf': [
-                {RESOURCE_SOURCE: {'type': 'string'}}
-            ],
-            'default-value': {'oneOf': [
-                {'type': 'string'},
-                {'type': 'number'},
-                {'type': 'boolean'}
-            ]}
-        },
         'oneOf': [
             {
-                'required': [RESOURCE_SOURCE]
+                'properties': {
+                    RESOURCE_SOURCE: {'type': 'string'},
+                    'default-value': {'oneOf': [
+                        {'type': 'string'},
```

Comment:
We can actually insert proper type in `lookup_type()` method.. (so if `string` is expected, `default-value` can only be `string`)

Reply:
That is a great idea and very clean!